src/mpid/ch4/src/ch4_startall.h

```diff
@@ -18,7 +18,7 @@ MPL_STATIC_INLINE_PREFIX int MPID_Startall(int count, MPIR_Request * requests[])
     for (i = 0; i < count; i++) {
         MPIR_Request *const preq = requests[i];
         /* continue if the source/dest is MPI_PROC_NULL */
-        if (MPIDIG_REQUEST(preq, rank) == MPI_PROC_NULL)
+        if (MPIDI_PREQUEST(preq, rank) == MPI_PROC_NULL)
```

Comment:
This fix is good, but (you'll hear this from me a lot) the commit message needs to be improved. It only describes what you did. You should describe **why** you did it. In this case, there was a user-observable bug #5053.

FWIW, I was able to write a test program that hung on this bug when it shouldn't have:
```
#include <mpi.h>
#include <stdlib.h>
#include <stdio.h>

int main(void) {
    MPI_Init(NULL, NULL);

    MPI_Request request = MPI_REQUEST_NULL;
    MPI_Recv_init(NULL, 0, MPI_INT, MPI_PROC_NULL, 0, MPI_COMM_WORLD, &request);
    MPI_Start(&request);
    MPI_Wait(&request, MPI_STATUS_IGNORE);
    MPI_Request_free(&request);

    MPI_Finalize();

    return 0;
}
```